packages/serverless/test/awslambda.test.ts

```diff
@@ -0,0 +1,378 @@
+// NOTE: I have no idea how to fix this right now, and don't want to waste more time, as it builds just fine — Kamil
+// eslint-disable-next-line import/no-unresolved
+import { Event } from '@sentry/types';
+import { Callback, Handler } from 'aws-lambda';
+
+import * as Sentry from '../src';
+
+const { wrapHandler } = Sentry.AWSLambda;
+
+/**
+ * Why @ts-ignore some Sentry.X calls
+ *
+ * A hack-ish way to contain everything related to mocks in the same __mocks__ file.
+ * Thanks to this, we don't have to do more magic than necessary. Just add and export desired method and assert on it.
+ */
+
+// Default `timeoutWarningLimit` is 500ms so leaving some space for it to trigger when necessary
+const DEFAULT_EXECUTION_TIME = 100;
+const fakeEvent = {
+  fortySix: 'o_O',
+};
+const fakeContext = {
+  callbackWaitsForEmptyEventLoop: false,
+  functionName: 'functionName',
+  functionVersion: 'functionVersion',
+  invokedFunctionArn: 'invokedFunctionArn',
+  memoryLimitInMB: 'memoryLimitInMB',
+  awsRequestId: 'awsRequestId',
+  logGroupName: 'logGroupName',
+  logStreamName: 'logStreamName',
+  getRemainingTimeInMillis: () => DEFAULT_EXECUTION_TIME,
+  done: () => {},
+  fail: () => {},
+  succeed: () => {},
+  ytho: 'o_O',
+};
+const fakeCallback: Callback = (err, result) => {
+  if (err === null || err === undefined) {
+    return result;
+  }
+  return err;
+};
+
+describe('AWSLambda', () => {
+  afterEach(() => {
+    // @ts-ignore see "Why @ts-ignore" note
+    Sentry.resetMocks();
+  });
+
+  describe('wrapHandler() options', () => {
+    test('flushTimeout', async () => {
+      expect.assertions(1);
+
+      const error = new Error('wat');
+      const handler = () => {
+        throw error;
+      };
+      const wrappedHandler = wrapHandler(handler, { flushTimeout: 1337 });
+
+      try {
+        await wrappedHandler(fakeEvent, fakeContext, fakeCallback);
+      } catch (e) {
+        expect(Sentry.flush).toBeCalledWith(1337);
+      }
+    });
+
+    test('rethrowAfterCapture', async () => {
+      expect.assertions(3);
+
+      const error = new Error('wat');
+      const handler = () => {
+        throw error;
+      };
+      const wrappedHandlerWithRethrow = wrapHandler(handler, { rethrowAfterCapture: true });
+      const wrappedHandlerWithoutRethrow = wrapHandler(handler, { rethrowAfterCapture: false });
+
+      try {
+        await wrappedHandlerWithRethrow(fakeEvent, fakeContext, fakeCallback);
+      } catch (e) {
+        expect(e).toBe(error);
+        expect(Sentry.flush).toBeCalledTimes(1);
+      }
+
+      await wrappedHandlerWithoutRethrow(fakeEvent, fakeContext, fakeCallback);
+      expect(Sentry.flush).toBeCalledTimes(2);
```

Comment:
Logaf: M

Is there a negative assertion you could add here, to prove that the error was not rethrown? Maybe something like 

```suggestion
      expect(await wrappedHandlerWithoutRethrow(fakeEvent, fakeContext, fakeCallback)).not.toThrow();
      expect(Sentry.flush).toBeCalledTimes(2);
```
?

(I get that it's not wrapped in a try-catch, and therefore the test itself would error if it didn't work, but it's easier to literally just read that in the test.)